axelrod/tests/test_round_robin.py

```diff
@@ -46,6 +50,22 @@ def test_deterministic_cache(self):
         self.assertEqual(rr.deterministic_cache[(axelrod.Cooperator, axelrod.Defector)], (0, 100))
         self.assertFalse((axelrod.Random, axelrod.Random) in rr.deterministic_cache)
 
+    def test_noisy_cache(self):
```

Comment:
the cache should not be updated if the tournament is noisy. this test checks that the cache is still empty after playing a noisy tournament even with otherwise deterministic players.